### DISCUSSION TITLE
Google Adsense - How to enable http without enforcing https

### DISCUSSION THREAD

**Discussion by initeshjain** [vercel/community#2718] [Help] [answered]
### Summary

I am having issue where google is unable to find ads.txt file on my site. I believe google is trying to access it on http and hence vercel enforce https, google received 302 permanent redirects.

How to solve this? I want to use http and https both without redirecting.

### Steps to Reproduce

Create a new next js app and place a ads.txt file with dummy data in public folder.
deploy it on vercel.

try to access the ads.txt file by powershell: 
invoke-restmethod -uri <your-vercel-url>/ads.txt

you will get 302 permanent redirect.

**Comment by smaeda-ks** [comment]
@initeshjain Google should be able to follow redirects. What's the exact URL to the file? I can check

> I am having issue where google is unable to find ads.txt file on my site

What's the exact error you see?

**Reply by initeshjain** [reply]
Hi @smaeda-ks, https://[noobgeek.in/ads.txt](https://noobgeek.in/ads.txt) this is the URL with https.
but when I run it in powershell as (without https):

Invoke-RestMethod noobgeek.in/ads.txt, I get the following error:
![image](https://github.com/vercel/community/assets/47496910/024790e9-4f2f-42ec-ab5e-333ba7eb47aa)


So I deployed same NextJs code on netlify and tried it. It is working.
Invoke-RestMethod rainbow-daffodil-991b3e.netlify.app/ads.txt
![image](https://github.com/vercel/community/assets/47496910/2ec07833-cfeb-4a80-8cd7-3e8121372e28)

**Reply by smaeda-ks** [reply, marked as answer]
@initeshjain I'm talking about Google's error, not from your Powershell command. The error you're mentioning so far is just a problem with Powershell in general. They treat non-200 status code as an error (including 3xx), and that's wrong anyway. Having http->https redirect doesn't block Google from crawling your website. They can handle it just fine, and that's not the case here.

Please note that there was an issue recently fixed by Google:
https://github.com/orgs/vercel/discussions/2611#discussioncomment-6047511

**Comment by initeshjain** [comment]
@smaeda-ks sorry my bad. I was getting the error in Adsense console as ads.txt is not detected.
But 1 hr back it is detected and authorised.

I am closing this thread.
